pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,273 @@
+    """
+    Creates an array of frequencies.
+
+    ..todo:: Implement log scaling
+
+    Parameters
+    ----------
+    freq_start : float
+        Starting value for the frequency definition.
+    freq_stop: float
+        Stopping value for the frequency definition, inclusive.
+    freq_step: float, optional
+        Step value, for linearly spaced values between start and stop.
+
+    Returns
+    -------
+    freqs: 1d array
+        Frequency indices.
+    """
+    return np.arange(freq_start, freq_stop + freq_step, freq_step)
+
+
+def compute_wavelet_transform(
+    sig, freqs, fs=None, n_cycles=1.5, scaling=1.0, precision=10, norm=None
+):
+    """
+    Compute the time-frequency representation of a signal using morlet wavelets.
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    freqs : 1d array or list of float
+        If array, frequency values to estimate with morlet wavelets.
+        If list, define the frequency range, as [freq_start, freq_stop, freq_step].
+        The `freq_step` is optional, and defaults to 1. Range is inclusive of `freq_stop` value.
+    fs : float or None
+        Sampling rate, in Hz. Defaults to sig.rate if None is given.
+    n_cycles : float or 1d array
+        Length of the filter, as the number of cycles for each frequency.
+        If 1d array, this defines n_cycles for each frequency.
+    scaling : float
+        Scaling factor.
+    precision: int.
+        Precision of wavelet to use. Default is 8
+    norm : {None, 'sss', 'amp'}, optional
+        Normalization method:
+        * None - no normalization
+        * 'sss' - divide by the square root of the sum of squares
+        * 'amp' - divide by the sum of amplitudes
+
+    Returns
+    -------
+    mwt : 2d array
+        Time frequency representation of the input signal.
+
+    Notes
+    -----
+    This computes the continuous wavelet transform at specified frequencies across time.
+    """
+
+    if not isinstance(sig, (nap.Tsd, nap.TsdFrame, nap.TsdTensor)):
+        raise TypeError("`sig` must be instance of Tsd, TsdFrame, or TsdTensor")
+    if isinstance(n_cycles, (int, float, np.number)):
+        if n_cycles <= 0:
+            raise ValueError("Number of cycles must be a positive number.")
+
+    if isinstance(freqs, (tuple, list)):
+        freqs = _create_freqs(*freqs)
+
+    if fs is None:
+        fs = sig.rate
+
+    if isinstance(sig, nap.Tsd):
+        sig = sig.reshape((sig.shape[0], 1))
+        output_shape = (sig.shape[0], len(freqs))
+    else:
+        output_shape = (sig.shape[0], len(freqs), *sig.shape[1:])
+        sig = sig.reshape((sig.shape[0], np.prod(sig.shape[1:])))
+
+    mwt = np.zeros(
+        [sig.values.shape[0], len(freqs), sig.values.shape[1]], dtype=complex
+    )
+
+    filter_bank = _generate_morlet_filterbank(freqs, fs, n_cycles, scaling, precision)
+    for f_i, filter in enumerate(filter_bank):
```

Comment:
From here, if `filter_bank` is an array, you don't need to do all the reshaping. `sig.convolve(filter_bank, ep)` should work directly.

Reply:
It wont currently as the filter bank will be num_frequencies x length_of_frequency x 2(real and imag component) 
and convolve only works with 1 or 2 dimensional data

Reply:
time x wavepow in filter bank - conv once with real then with imag, then add